lib/publiccloud/gce.pm

```diff
@@ -170,14 +170,17 @@ sub cleanup {
 
     my $region = $self->{provider_client}->{region};
     my $project = $self->{provider_client}->{project_id};
-    my $instance_id = $self->get_terraform_output(".vm_name.value[0]");
+    my $query = ".vm_name.value[0]";
+    my $instance_id = $self->get_terraform_output($query);
     # gce provides full serial log, so extended timeout
-    if (!check_var('PUBLIC_CLOUD_SLES4SAP', 1) && defined($instance_id)) {
-        if ($instance_id =~ /$self->{resource_name}/) {
-            script_run("gcloud compute --project=$project instances get-serial-port-output $instance_id --zone=$region --port=1 > instance_serial.txt", timeout => 180);
+    if (!check_var('PUBLIC_CLOUD_SLES4SAP', 1)) {
+        if ($instance_id) {
+            my $res = script_run("gcloud compute --project=$project instances get-serial-port-output $instance_id --zone=$region --port=1 > instance_serial.txt", timeout => 180);
+            script_run("gcloud compute --project=$project instances get-serial-port-output $instance_id --port=1 -> instance_serial.txt", timeout => 180)
+              if ($res != 0);    # zone::region mismatch, retry without zone
```

Comment:
I am sorry but I don't understand this. Zone belongs to region[1]. What mismatch are you referring to?

[1]: https://cloud.google.com/compute/docs/regions-zones

Reply:
In our P.C. tests usually we set the Zone into REGION, therefore it works and we never noted errors.
But i.e. when like the SAP test that I cloned [14624544](https://openqa.suse.de/tests/14624544#settings), to experiment the routine, Region is `europe-west1` and not zone, when it comes to get serial log `gcloud compute --project=... instances get-serial-port-output 1 --zone=europe-west1 --port=1` the `--zone` is not ok and claims an [error](https://openqa.suse.de/tests/14676329#step/ssh_interactive_end/78) because it should be i.e. `europe-west1-1b`.
GCE only is using `zone` in this command.

Reply:
Holy moly, can I or will you create a ticket out of your message above?

I'd at least talk about it when Anton is back.

Reply:
Here is poo#[162620](https://progress.opensuse.org/issues/162620)

Reply:
https://progress.opensuse.org/issues/162620